### PR TITLE
Use summary table for performance and accuracy

### PR DESCRIPTION
**Pull request by prescod**

### CODE REVIEW

File: metaci/testresults/models.py [hunk 1]
```diff
@@ -274,3 +255,300 @@ class TestResultAsset(models.Model):
...
+                agg_failures=metrics["failures"],
+                agg_assertion_failures=metrics["assertion_failures"],
+                agg_DML_failures=metrics["DML_failures"],
+                agg_other_failures=metrics["other_failures"],
+            )
+        )
+        return method_contexts
+
+
+class TestResultPerfWeeklySummaryQuerySet(models.QuerySet):
```
Comment (davisagli) [inline metaci/testresults/models.py]: Weekly? Does this do the same thing as what is already in build.models?

File: metaci/testresults/models.py [hunk 2]
```diff
@@ -274,3 +255,300 @@ class TestResultAsset(models.Model):
...
+
+
+class TestResultPerfWeeklySummary(TestResultPerfSummaryBase):
+    """Weekly summary table/model"""
+
+    logger = logging.getLogger(__name__)
+    out_hdlr = logging.StreamHandler(sys.stdout)
+    out_hdlr.setLevel(logging.INFO)
+    logger.addHandler(out_hdlr)
+    logger.setLevel(logging.INFO)
```
Comment (davisagli) [inline metaci/testresults/models.py]: don't log to stdout

File: metaci/testresults/models.py [hunk 3]
```diff
@@ -274,3 +255,300 @@ class TestResultAsset(models.Model):
...
+        unique_together = ("rel_repo", "rel_branch", "rel_plan", "method", "week_start")
+        indexes = [
+            models.Index(fields=unique_together, name="testresult_weekly_lookup")
+        ]
+
+    week_start = models.DateField(null=False, blank=False)
+    objects = TestResultPerfWeeklySummaryQuerySet.as_manager()
+
+    @classmethod
+    def _parse_start_and_date_dates(cls, start_string, end_string):
```
Comment (davisagli) [inline metaci/testresults/models.py]: docstring

File: metaci/testresults/models.py [hunk 4]
```diff
@@ -274,3 +255,300 @@ class TestResultAsset(models.Model):
...
+            week_start + timezone.timedelta(days=7), use_tz=True
+        )
+
+        method_contexts = cls._get_queryset_for_dates(
+            week_start_with_timezone,
+            week_end_with_timezone,
+            week_start=Value(week_start, output_field=models.DateField()),
+        )
+
+        obsolete_objects = cls.objects.filter(week_start=week_start)
```
Comment (davisagli) [inline metaci/testresults/models.py]: comment

File: metaci/api/views/testresults.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        initial=["repo", "duration_average"],
+    )
+
+    ordering_fields = [(key, key) for (key, field) in includable_fields]
+
+    o = django_filters.rest_framework.OrderingFilter(fields=ordering_fields)
+    ordering_param_name = "o"
+
+
+class TestMethodResultListView(TestMethodPerfListView):
```
Comment (davisagli) [inline metaci/api/views/testresults.py]: Is this used?

File: metaci/testresults/models.py (hunk 1 above)
Comment (prescod) [reply metaci/testresults/models.py]: If this question is still open, please clarify it.

File: metaci/api/views/testresults.py (hunk 5 above)
Comment (prescod) [reply metaci/api/views/testresults.py]: e6f7e6a99b7137f7594d14311df5fc6ba21ea32e

File: metaci/testresults/models.py (hunk 4 above)
Comment (prescod) [reply metaci/testresults/models.py]: Part of 534c3404ab448173f73b8876f3d361b2642767bc

File: metaci/testresults/models.py (hunk 3 above)
Comment (prescod) [reply metaci/testresults/models.py]: Part of 534c3404ab448173f73b8876f3d361b2642767bc

File: metaci/testresults/models.py (hunk 2 above)
Comment (prescod) [reply metaci/testresults/models.py]: Part of 534c3404ab448173f73b8876f3d361b2642767bc

**Pull request merged**